### DISCUSSION TITLE
Addon Storybook 8 Migration: `Cannot read properties of null (reading 'useState')` in preview

### DISCUSSION THREAD

**Discussion by neil-morrison44** [storybookjs/storybook#26782] [Help]
### Summary

Hi,
I'm trying to migrate my addon which was working in storybook 7 to storybook 8 and am encountering an issue.

I've got the manager portion of it working, however the preview bit no longer seems to be rendered within a proper react context (or something, that seems to be the usual cause of the `Cannot read properties of null (reading 'useState')` / `Cannot read properties of null (reading 'useEffect')` issues).

My addon is React specific & the components I'm testing it with are React only.

Has preview changed to rendering things outside of the framework?
Is there a way to get it to render within the React tree?

Using storybook's own `useState` etc probably isn't an option since I need to set up React contexts anyway for the RecoilRoot bits & that's 3rd party stuff that expects to be in a "normal" React tree.

### Additional information

The branch with the changes is up at https://github.com/neil-morrison44/storybook-addon-recoil-flow/pull/7

Could be there's a better way of doing what I'm doing, essentially I need to wrap the React component in something which adds a sibling (`RecoilSnooper`) to it, as well as setting up `RecoilRoot` if it hasn't been already, then the `RecoilSnooper` communicates with the panel over the channels to show what's going on.

**Comment by SpookyJelly** [comment]
did you solved this problem?? I am stuck on same thing too
